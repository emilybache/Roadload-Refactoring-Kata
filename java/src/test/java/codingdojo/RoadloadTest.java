package codingdojo;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class RoadloadTest {
    @Test
    void calculateCO2Values() {

        CombinationApprovals.verifyAllCombinations(this::doCalculateCO2Values,
                new Boolean[]{false, true},
                new Boolean[]{false, true},
                new Double[]{3.5, null},
                new Double[]{3.2, null},
                new Double[]{3.1, null}
        );
    }

    private String doCalculateCO2Values(Boolean isMidPoint, Boolean isSinglePoint,
                                        Double highRoadloadF0, Double midRoadloadF0, Double lowRoadloadF0) {
        var values = new CO2Values();
        values.High_roadload_f0 = highRoadloadF0;
        values.High_roadload_f1 = 10.6575;
        values.High_roadload_f2 = 6.7;
        values.High_roadload_tm = 0.003;
        values.Mid_roadload_f0 = midRoadloadF0;
        values.Mid_roadload_f1 = 9.75;
        values.Mid_roadload_f2 = 5.9;
        values.Mid_roadload_tm = 0.0025;
        values.Low_roadload_f0 = lowRoadloadF0;
        values.Low_roadload_f1 = 8.45;
        values.Low_roadload_f2 = 5.3;
        values.Low_roadload_tm = 0.002;
        values.IsMidPoint = isMidPoint;
        values.IsSinglePoint = isSinglePoint;

        CO2Calculator.calculateCO2Values(values);

        return String.format("(%s, %s, %s)", values.Low_roadload_combined, values.Mid_roadload_combined, values.High_roadload_combined);
    }

}
