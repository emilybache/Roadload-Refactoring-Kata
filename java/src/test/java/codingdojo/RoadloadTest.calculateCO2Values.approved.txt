[false, false, 3.5, 3.2, 3.1] => (4.222231047218139E-4, null, 7.987578796146626E-4) 
[false, false, 3.5, 3.2, null] => (null, null, 7.987578796146626E-4) 
[false, false, 3.5, null, 3.1] => (4.222231047218139E-4, null, 7.987578796146626E-4) 
[false, false, 3.5, null, null] => (null, null, 7.987578796146626E-4) 
[false, false, null, 3.2, 3.1] => (4.222231047218139E-4, null, null) 
[false, false, null, 3.2, null] => (null, null, null) 
[false, false, null, null, 3.1] => (4.222231047218139E-4, null, null) 
[false, false, null, null, null] => (null, null, null) 
[false, true, 3.5, 3.2, 3.1] => (null, null, 7.987578796146626E-4) 
[false, true, 3.5, 3.2, null] => (null, null, 7.987578796146626E-4) 
[false, true, 3.5, null, 3.1] => (null, null, 7.987578796146626E-4) 
[false, true, 3.5, null, null] => (null, null, 7.987578796146626E-4) 
[false, true, null, 3.2, 3.1] => (null, null, null) 
[false, true, null, 3.2, null] => (null, null, null) 
[false, true, null, null, 3.1] => (null, null, null) 
[false, true, null, null, null] => (null, null, null) 
[true, false, 3.5, 3.2, 3.1] => (4.222231047218139E-4, 5.985347985347986E-4, 7.987578796146626E-4) 
[true, false, 3.5, 3.2, null] => (null, 5.985347985347986E-4, 7.987578796146626E-4) 
[true, false, 3.5, null, 3.1] => (4.222231047218139E-4, null, 7.987578796146626E-4) 
[true, false, 3.5, null, null] => (null, null, 7.987578796146626E-4) 
[true, false, null, 3.2, 3.1] => (4.222231047218139E-4, 5.985347985347986E-4, null) 
[true, false, null, 3.2, null] => (null, 5.985347985347986E-4, null) 
[true, false, null, null, 3.1] => (4.222231047218139E-4, null, null) 
[true, false, null, null, null] => (null, null, null) 
[true, true, 3.5, 3.2, 3.1] => (null, null, 7.987578796146626E-4) 
[true, true, 3.5, 3.2, null] => (null, null, 7.987578796146626E-4) 
[true, true, 3.5, null, 3.1] => (null, null, 7.987578796146626E-4) 
[true, true, 3.5, null, null] => (null, null, 7.987578796146626E-4) 
[true, true, null, 3.2, 3.1] => (null, null, null) 
[true, true, null, 3.2, null] => (null, null, null) 
[true, true, null, null, 3.1] => (null, null, null) 
[true, true, null, null, null] => (null, null, null) 
